valhalla/sif/edgelabel.h

```diff
@@ -72,19 +73,21 @@ class EdgeLabel {
             const bool closure_pruning,
             const bool has_measured_speed,
             const InternalTurn internal_turn,
-            const uint8_t path_id = 0)
+            const uint8_t path_id = 0,
+            const bool destonly = false)
```

Comment:
we could instead pass the hgv flag here and then edgelabel would be the only place that has to have logic to decide whther to use the regular dest only flag or the hgv flag instead of this ternary/if leaking out into the algorithms

Reply:
unless we require the caller to differentiate wrt direction. I think dijkstras is the only not doing that yet, it'd be trivial to refactor that. eventually we should bake that info into the EdgeLabel, but for now that'd work and still be nicer than having every algo care about it.

Reply:
meeh, forget about the previous comment. of course every direction puts the _expansion_ edge into the label, for reverse search we want the opposite. so unfortunately we have to leak it into the algos at this point.